python/requirements.txt

```diff
@@ -0,0 +1,4 @@
+numpy==1.16.4
+scipy==1.2.2
+pandas==0.23.2
+pyarrow==0.15.1
```

Comment:
@dongjoon-hyun how about we put this in https://github.com/apache/spark/blob/master/dev/requirements.txt since these are optional dependencies (and we can say it's needed for the dev?)

```
# ML/MLlib
numpy>=1.7
scipy>=1.2.2
# SQL
pandas>=0.23.2
pyarrow>=0.15.1
```

Reply:
It conventionally follows `setup.py` version specification (https://github.com/apache/spark/blob/master/python/setup.py#L216-L220).

Alternatively, we can place a separate file named `requirements-dev.txt` under `python` folder (since `python` folder itself can be considered as a separate independent Python project).

Reply:
Sure. Thanks. I'll do the followings.
1. Move to `dev`.
2. Test with numpy==1.7 and use that version. (I saw it in the setup.py, but last time, it didn't work for me.)

BTW, I'll use `==` instead of `>=` since it's risky (especially for `pyarrow`).
That's the reason why this PR title is `minimum` instead of `valid range`.

Reply:
Oh. I rename this to `requirements-dev.txt` under `python` according to the second option.